context/app/static/js/components/workspaces/api.ts

```diff
@@ -256,11 +272,38 @@ export function useStartWorkspace() {
   return { startWorkspace, isStartingWorkspace: isMutating };
 }
 
+interface CreateWorkspaceBody {
+  name: string;
+  description: string;
+  workspace_details: {
+    globus_groups_token: string;
+    files: {
+      name: string;
+      content?: string;
+    }[];
+    symlinks: {
+      name: string;
+      dataset_uuid: string;
+    }[];
+  };
+}
+
 interface CreateWorkspaceArgs extends APIAction {
-  body: unknown;
+  body: CreateWorkspaceBody;
 }
 
 async function createWorkspaceFetcher(_key: string, { arg: { body, url, headers } }: { arg: CreateWorkspaceArgs }) {
+  trackEvent({
+    category: 'Workspace Creation',
+    action: 'Create Workspace',
+    value: {
+      name: body.name,
+      description: body.description,
+      globus_groups_token: body.workspace_details.globus_groups_token,
+      files: body.workspace_details.files.map((f) => f.name),
```

Comment:
Should we track workspace template names and uuids instead of the files? The files can include the templates strings which could be quite large/uninformative.

Reply:
Realizing now that's already in place!